lib/src/core/provider/power_off_provider.dart

```diff
@@ -21,26 +21,41 @@ class PowerOffProvider with ChangeNotifier {
   ///.....
   int? city = -1;
 
-  final MockRepository _mockRepository;
+  final MockRepository? _mockRepository;
 
-  late List<Set<Marker>> _markers;
+  List<Set<Marker>>? _markers;
 
-  late List<DateTime> _dates;
+  List<DateTime>? _dates;
 
-  UnmodifiableListView<Set<Marker>> get markers =>
-      UnmodifiableListView<Set<Marker>>(_markers);
+  UnmodifiableListView<Set<Marker>>? get markers =>
+      UnmodifiableListView<Set<Marker>>(_markers!);
 
-  UnmodifiableListView<DateTime> get dates =>
-      UnmodifiableListView<DateTime>(_dates);
+  UnmodifiableListView<DateTime>? get dates =>
+      UnmodifiableListView<DateTime>(_dates!);
 
   Future<void> init() async {
     //will be replaced with some method which will generate markers and dates from retrieved data
-    _markers = await _mockRepository.getMarkers(
+    _markers = await _mockRepository!.getMarkers(
+        //TODO: change for real amount of markers
         iconForMap: await _convertingIconIntoBytes());
-    _dates = await _mockRepository.getDates();
+    _dates = await _mockRepository!.getDates();
   }
 
   Future<BitmapDescriptor> _convertingIconIntoBytes() async {
+    final dateTimeNow = DateTime.now();
+
+    final date = await _mockRepository!.getDates();
+    //_dates![index!];
+
+    ///[Colors.red] no electricity
+    ///[Colors.yellow] no electricity soon
+    ///[Colors.green] electricity is on
+//TODO: fix index and dateMock
+    Color? iconColor;
+    if (date.elementAt(0).isAtSameMomentAs(dateTimeNow)) iconColor = Colors.red;
+    if (date.elementAt(0).isBefore(dateTimeNow)) iconColor = Colors.yellow;
+    if (date.elementAt(0).isAfter(dateTimeNow)) iconColor = Colors.green;
```

Comment:
Better create variable to get the first element using `first` getter from `List` class and use `if else` statement instead of single `if` myltiple times